docs/proposals/network/README.md

```diff
@@ -0,0 +1,67 @@
+# Network
+
+## How it works
+
+### Introduction
+
+The objective of the network service is to make several instances communicate with each other as if they were all connected to a single private network, no matter if they are on the same machine or not.
+
+### Suggested approach
+
+To try to solve this problem, the network service is responsible for creating, configuring and deleting virtual network interfaces used to connect instances to each other within a cluster.
+
+This service has to be running on all nodes of the cluster.
+
+When adding a new node to a cluster, a call to the network service must be done to set up a new container network interface (CNI). A CNI links the virtual interfaces associated with the instances running on that node to the rest of the cluster.
+
+A call to the network service must also be made before creating a new instance in order to set up the virtual network interface connecting the instance to a CNI.
+
+After the destruction of an instance, the network service must be called to remove the network interface associated with it and reconfigure the routing tables.
+
+![Node example](schema.png)
+
+## API
+
+### Messages
+
+```protobuf
+// Request structure used to create a new virtual interface
```

Comment:
By `virtual interface` you actually mean a `virtual network interface` for a given instance.

In order to do so, you will have to pass the workload identifier as part of the request. I'm assuming it's going to be the network service responsibility to set up that interface on the host and in the instance networking namespace. So in order to do that the network service will have to obviously identify the instance netns, typically through a unique instance identifier.